src/main/java/com/aws/greengrass/clientdevices/auth/ClientDevicesAuthService.java

```diff
@@ -200,19 +220,54 @@ protected void install() throws InterruptedException {
 
             if (whatHappened == WhatHappened.initialized || node == null) {
                 updateDeviceGroups(whatHappened, deviceGroupTopics);
-                updateCAType(caTypeTopic);
+                updateCertificateAuthority();
             } else if (node.childOf(DEVICE_GROUPS_TOPICS)) {
                 updateDeviceGroups(whatHappened, deviceGroupTopics);
-            } else if (node.childOf(CA_TYPE_TOPIC)) {
-                if (caTypeTopic.getOnce() == null) {
+            } else if (node.childOf(CERTIFICATE_AUTHORITY_TOPIC)) {
+                if (certificateAuthorityTopic.getOnce() == null) {
                     return;
                 }
-                updateCAType(caTypeTopic);
+                updateCertificateAuthority();
             }
         });
 
     }
 
+    private void updateCertificateAuthority() {
+        String caKeyPath = Coerce.toString(config.findOrDefault("", CERTIFICATE_AUTHORITY_TOPIC, CA_PRIVATE_KEY_URI));
+        String caCertPath =
+                Coerce.toString(config.findOrDefault("", CERTIFICATE_AUTHORITY_TOPIC, CA_CERTIFICATE_URI));
+        Topic caType = config.lookup(CERTIFICATE_AUTHORITY_TOPIC, CA_TYPE_TOPIC);
+
+        // If the key and cert path of CA are empty, generate a CA
+        if (Utils.isEmpty(caKeyPath) || Utils.isEmpty(caCertPath)) {
+            updateCAType(caType);
+        } else {
+            // Get CA from URI and upload
+            try {
+                URI privateKeyUri = new URI(caKeyPath);
+                URI certificateUri = new URI(caCertPath);
+                KeyPair keyPair = securityService.getKeyPair(privateKeyUri, certificateUri);
+                List<X509Certificate> certificateChain = securityService.getCertificateChain(privateKeyUri,
+                        certificateUri);
+                certificateManager.setCA(keyPair.getPrivate(), (Certificate[]) certificateChain.toArray());
+            } catch (URISyntaxException e) {
+                throw new ServiceError("Exception occurred while parsing the certificate authority "
+                        + "configuration");
+            } catch (ServiceUnavailableException | KeyLoadingException | IOException | CertificateException e) {
```

Comment:
ServiceUnavailableException must be retried, you cannot just bail out. It is expected to be thrown for a while, before the pkcs11 provider is loaded.